mlir/lib/Dialect/Arith/Transforms/ExpandOps.cpp

```diff
@@ -23,6 +26,16 @@ namespace arith {
 
 using namespace mlir;
 
+static Value createFloatConst(Location loc, Type type, float value,
+                              PatternRewriter &rewriter) {
+  auto attr = rewriter.getFloatAttr(getElementTypeOrSelf(type), value);
+  if (auto shapedTy = dyn_cast<ShapedType>(type)) {
+    return rewriter.create<arith::ConstantOp>(
```

Comment:
We can update the `attr` here: `attr = DenseElementsAttr::get(shapedTy, attr)`. It will return the right thing. (Both are fine to me).